lldb/source/Plugins/Process/amdgpu-core/ProcessAmdGpuCore.cpp

```diff
@@ -0,0 +1,348 @@
+//===-- ProcessAmdGpuCore.cpp
+//------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdlib>
+
+#include <memory>
+
+#include "Plugins/DynamicLoader/GpuCore-DYLD/DynamicLoaderGpuCoreDYLD.h"
+#include "Plugins/ObjectFile/ELF/ObjectFileELF.h"
+#include "lldb/Core/Debugger.h"
+#include "lldb/Core/Module.h"
+#include "lldb/Core/PluginManager.h"
+#include "lldb/Target/DynamicLoader.h"
+#include "lldb/Utility/DataExtractor.h"
+#include "lldb/Utility/LLDBLog.h"
+#include "lldb/Utility/Log.h"
+#include "llvm/ADT/ScopeExit.h"
+#include "llvm/Support/Threading.h"
+
+#include "ProcessAmdGpuCore.h"
+#include "ThreadAMDGPU.h"
+
+using namespace lldb_private;
+using namespace lldb;
+
+LLDB_PLUGIN_DEFINE(ProcessAmdGpuCore)
+
+llvm::StringRef ProcessAmdGpuCore::GetPluginDescriptionStatic() {
+  return "ELF amd gpu core dump plug-in.";
+}
+
+void ProcessAmdGpuCore::Initialize() {
+  static llvm::once_flag g_once_flag;
+
+  llvm::call_once(g_once_flag, []() {
+    // Register as GPU Process plugin (for merged CPU+GPU cores)
+    // AMD only supports merged mode, not standalone GPU-only cores
+    PluginManager::RegisterGpuProcessPlugin(
+        GetPluginNameStatic(), GetPluginDescriptionStatic(), CreateInstance);
+  });
+}
+
+void ProcessAmdGpuCore::Terminate() {
+  PluginManager::UnregisterGpuProcessPlugin(ProcessAmdGpuCore::CreateInstance);
+}
+
+lldb::ProcessSP ProcessAmdGpuCore::CreateInstance(lldb::TargetSP target_sp,
+                                                  lldb::ListenerSP listener_sp,
+                                                  const FileSpec *crash_file,
+                                                  bool can_connect) {
+  return std::make_shared<ProcessAmdGpuCore>(target_sp, listener_sp,
+                                             *crash_file);
+}
+
+bool ProcessAmdGpuCore::CanDebug(lldb::TargetSP target_sp,
+                                 bool plugin_specified_by_name) {
+  // AMD GPU core plugin only supports merged CPU+GPU cores
+  // This is only called from ProcessElfGpuCore::LoadGpuCore()
+  // Since we're registered as GPU Process plugin, never called from
+  // normal Process::FindPlugin() discovery
+  auto cpu_process = GetCpuProcess();
+  assert(cpu_process && "how can we not have a CPU process in CPU+GPU cores?");
+
+  // Check if this core file has AMD GPU notes (type 33 =
+  // NT_AMDGPU_KFD_CORE_STATE)
+  return cpu_process->GetAmdGpuNote().has_value();
```

Comment:
This is the kind of work that needs to be done in `CreateInstance` and isn't needed as a separate function like `ProcessAmdGpuCore::CanDebug`